### PR TITLE
fifth increment

### PR DESCRIPTION
**Pull request by volnistii11**

### CODE REVIEW

File: internal/app/storage/storage.go [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package storage
+
+var URLMap map[string]string
```
Comment (arttet) [inline internal/app/storage/storage.go:3]: Which rule of `SOLID` did you break?

File: internal/app/handlers/handler.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+	"github.com/volnistii11/URL-shortener/internal/app/config"
+	"github.com/volnistii11/URL-shortener/internal/app/storage"
+	"math/rand"
+	"net/http"
+	"time"
+)
+
+func init() {
+	rand.Seed(time.Now().UnixNano())
+	storage.URLMap = map[string]string{}
```
Comment (arttet) [inline internal/app/handlers/handler.go:15]: You shouldn't do it here. Please use `struct` or `function`

**Review by arttet — CHANGES_REQUESTED** [review]
Good job!

File: internal/app/handlers/handler.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+func randString(n int) string {
+	b := make([]rune, n)
+	for i := range b {
+		b[i] = letters[rand.Intn(len(letters))]
+	}
+	return string(b)
+}
+
+func errorResponse(err string) gin.H {
```
[the remark this answers is not in the document]
Comment (volnistii11) [reply internal/app/handlers/handler.go:68]: Sry, what do you mean?)

File: internal/app/storage/storage.go (hunk 1 above)
Comment (volnistii11) [reply internal/app/storage/storage.go:3]: Type inside a var name?

File: internal/app/handlers/handler.go [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+	ctx.String(http.StatusCreated, "%v%v", respondingServerAddress, shortURL)
+}
+
+func GetFullURL(ctx *gin.Context) {
+	shortURL := ctx.Params.ByName("short_url")
+
+	s := storage.GetStorage()
+	fullURL, err := s.ReadURL(shortURL)
+
+	if err != nil {
```
Comment (arttet) [inline internal/app/handlers/handler.go:49]: Nit: you can reduce this space
```
	if err != nil {
		ctx.JSON(http.StatusBadRequest, errorResponse(err))
                return
	}

        ctx.Header("Location", fullURL)
        ctx.Status(http.StatusTemporaryRedirect)
```

**Review by arttet — APPROVED** [review]
Well done!

**Pull request merged**